packages/editor/src/components/local-autosave-monitor/index.js

```diff
@@ -112,16 +118,32 @@ function useAutosaveNotice() {
 				],
 			}
 		);
-	}, [ postId ] );
+	}, [ isEditedPostNew, postId ] );
+
+	// Once the isEditedPostNew changes from true to false, let's clear the auto-draft autosave.
+	const wasEditedPostNew = usePrevious( isEditedPostNew );
+	const prevPostId = usePrevious( postId );
+	useEffect( () => {
+		if ( prevPostId === postId && wasEditedPostNew && ! isEditedPostNew ) {
+			localAutosaveClear( postId, true );
+		}
+	}, [ isEditedPostNew, postId ] );
```

Comment:
Should this be moved to `useAutosavePurge` instead? Seems like this hooks is meant to be used for these things?